Add operator breaking changes documentation (#7753)
stackrox/stackrox: operator/EXTENDING_CRDS.md

* Nullability: See [Kubernetes Documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting-and-nullable).
* Late initialization: See [API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#late-initialization).
* Labels, Selector and Annotations: See [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#label-selector-and-annotation-conventions).

> ## Breaking changes

Reviewer: Perhaps this should even go before the "Style recommendations" section

Author: Agree, done

> The CR of an operator acts like a public API to the ACS configuration.

Reviewer: ```suggestion
The CR of an operator is the public API of the ACS configuration.
```

Author: done

Additionally we need to keep in mind that CRs are often managed in CI/CD pipelines which would 
break existing automations.

 * Never remove a CR field from a CRD
 * Never remove an enum field from a CRD

>  * Defined values must continue to stay valid values, i.e.: 

Reviewer: ```suggestion
 * Defined values must continue to stay valid values, e.g.: 
```

Author: done

   * `KERNEL_MODULE` defaults to `EBPF`, but `KERNEL_MODULE` remains a valid value as an alias for `EBPF`

>    * central-db is still a valid config in the Central CR even though users must use postgres

Reviewer: ```suggestion
   * `central-db` is still a valid config section in the Central CR even though users must use postgres
```

Author: done

 * Defaults can change when the change is not going to break things

**Introducing a breaking change:**

This is possible by promoting a CR, i.e. from `v1alpha1` to `v1beta`. Technically this 

> is equally to introducing moving from a `/v1` to a `/v2` API.

Reviewer: ```suggestion
is equal to moving from a `/v1` to a `/v2` API.
```

Author: done
